src/components/page/AppPage.js

```diff
@@ -47,34 +48,38 @@ class AppPage extends React.Component {
     onSpinner = () => {
         this.setState({
             clicked: true
-        })
+        });
+        
+        this.context.toggleRedirect();
 
         setTimeout(() => {
-            window.location.href = '/quiz'
-        }, 750)
+            this.props.history.push('/quiz')
+        }, 750);
```

Comment:
Why don't you set time randomly from 500-1000?!

Reply:
Done!